### ISSUE TITLE
client reset  observe failed

### ISSUE THREAD

**Issue by ghostyu** [mcollina/node-coap#154]
in client side, i want to send RST packet to server.
`req.on('response', function(res) {
  console.log("---------------------response---------------------");
  console.log(res.headers);

  var rcvCount = 5;
  res.on('data', function(data) {
    console.log(data);
    rcvCount--
    if(rcvCount == 0){
      //it will throw new Error: Empty messages must be empty
      req.reset();
    }
  });
})
req.end()`

when i call req.reset(), it will throw Empty messages must be empty error

**Comment by Pluuuto** [comment]
I have almost the same problem. When I am using "res.reset()" at server-end.
```
const coap = require('coap')

const server = coap.createServer()

server.on('request', (req, res) => {
  console.log(`Receive request: ${req.url}`)
  res.reset()
})

server.listen(5683, '127.0.0.1')

```

It will cause client-end throwing "Empty messages must be empty error". Using wireshark to capture and I find server-end sending NON msg with '5.00 Internal Server Error' and body 'Empty messages must be empty' after send RST msg. 

![image](https://user-images.githubusercontent.com/9493652/32530723-61d3dd76-c405-11e7-8f81-f2b8790da683.png)

NON msg has the same token with the NON request which client-end sent. I think this cause client-end throws error. I don't know why coap source code deal with reset on this way, maybe it's a bug? (reference to 'fastseries' lib)

**Comment by Pluuuto** [comment]
I found this is a bug because the RST msg has token. RFC7252-4.1 specific as follows:

![image](https://user-images.githubusercontent.com/9493652/32531104-a610128c-c407-11e7-8430-dad554598d56.png)

The Error "Empty messages must be empty" is thrown by 'coap-packet' lib 'index.js':

![image](https://user-images.githubusercontent.com/9493652/32531257-5f9894c2-c408-11e7-9c68-b5fe12ae6d74.png)

And I check coap source code 'outgoing_message.js' and I find this:

![image](https://user-images.githubusercontent.com/9493652/32531284-896637aa-c408-11e7-8347-b4b33d1bc8d0.png)

**packet.token is NOT set to null**! I think this is the essential problem. After I add "packet.token = ''" at the following, the client-end doesn't thow the Error.

![image](https://user-images.githubusercontent.com/9493652/32531374-09440c9a-c409-11e7-9f3c-68f3bf13cdd7.png)

**Comment by GiedriusM** [comment]
@Pluuuto, I agree that this is a bug. Moreover, there seems to be more (un?)related problems that come out to the surface. There's also some difference depending if the client is on the same host as the server or not. Removing the token is a good first step, which fixes the stated problem, however as a side effect a program with a single coap request does not terminate (which may be another bug). If you want, you can submit a PR with the fix and tests that cover the scenario. If not I'll do it once I get to take a better look at all of the problems that this bug revealed.

**Comment by GiedriusM** [comment]
Please check if current master branch has this issue anymore.

**Comment by GiedriusM** [comment]
@ghostyu, @Pluuuto are we good to close this issue?

**Comment by robertbbb** [comment]
I seem to be having the same problem. Is there a branch where this bug was fixed?

**Comment by lucarv** [comment]
The issue exists in 0.23.1. I get the same error....
/app/node_modules/coap-packet/index.js:102
      throw new Error('Empty messages must be empty')

**Comment by JKRhb** [comment]
@robertbbb @lucarv Could you give a bit more context under which conditions the error occured?

**Comment by cbiale** [comment]
This issue  continue in the release 1.3.0. Any possibility of a solution or how to send an RST to the server in the case of observing. Thanks.
